docs/4.0-migration-guide/pake.md

```diff
@@ -1,5 +1,111 @@
+
+### Combine `psa_pake_set_password_key()` with `psa_pake_setup()`
+
+The function `psa_pake_set_password_key()` has been removed. Its `key` argument is now passed to `psa_pake_setup()` which has gained a new `key` parameter.
+
+Before:
+
+```
+status = psa_pake_setup(&operation, &cipher_suite);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+status = psa_pake_set_password_key(&operation, key);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+```
+
+Now:
+
+```
+status = psa_pake_setup(&operation, key, &cipher_suite);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+```
+
+### Move the hash algorithm parameter into the algorithm identifier
+
+The function `psa_pake_cs_set_hash()` has been removed. Its `hash` argument is now passed to `PSA_ALG_JPAKE()` which is now a function-like macro with one parameter.
+
+Before:
+
+```
+psa_pake_cs_set_algorithm(&cipher_suite, PSA_ALG_JPAKE);
+psa_pake_cs_set_hash(&cipher_suite, PSA_ALG_SHA_256);
+```
+
+Now:
+
+```
+psa_pake_cs_set_algorithm(&cipher_suite, PSA_ALG_JPAKE(PSA_ALG_SHA_256));
+```
+
+To check if a given algorithm is J-PAKE, the new `PSA_ALG_IS_JPAKE()` macro has been added.
+
+Before: `if (alg == PSA_ALG_JPAKE)`
+
+Now: `if (PSA_ALG_IS_JPAKE(alg))`
+
+The function `psa_pake_cs_get_hash()` has also been removed.
+
+### Replace `psa_pake_get_implicit_key()` with `psa_pake_get_shared_key()`
+
+The function `psa_pake_get_implicit_key()`, which injects the shared secret into a key derivation operation, has been removed. Its replacement is `psa_pake_get_shared_key()` which stores the shared secret in a new key. That new key can then be used as part of a key derivation operation.
+
+Before:
+
+```
+status = psa_pake_get_implicit_key(&pake_op, &derivation_op);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+```
+
+Now:
+
+```
+psa_algorithm_t alg = PSA_ALG_TLS12_PSK_TO_MS(PSA_ALG_SHA_256);
+                      // or other key derivation algorithm
+psa_key_id_t shared_key_id = (psa_key_id_t) 0;
+
+psa_key_attributes_t shared_key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+psa_set_key_usage_flags(&shared_key_attributes, PSA_KEY_USAGE_DERIVE);
+psa_set_key_algorithm(&shared_key_attributes, alg);
+psa_set_key_type(&shared_key_attributes, PSA_KEY_TYPE_DERIVE);
+
+status = psa_pake_get_shared_key(&pake_op,
+                                 &shared_key_attributes,
+                                 &shared_key_id);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+
+status = psa_key_derivation_input_key(&derivation_op,
+                                      PSA_KEY_DERIVATION_INPUT_SECRET,
+                                      shared_key_id);
+if (status != PSA_SUCCESS) // error handling ommited for brevity
+
+psa_reset_key_attributes(&shared_key_attributes);
+psa_destroy_key(shared_key_id);
```

Comment:
Can this be done immediately, or do you need to wait until the key derivation operation is finished?

(Answer: according to the PSA specification, you must not destroy a key while it's in use in an operation: that would effectively abort the operation. But in our implementation, destroying a key while an operation is in progress is currently undetected.)